packages/fxa-auth-server/lib/push.js

```diff
@@ -371,93 +350,131 @@ module.exports = function (log, db, config) {
      */
     async sendPush(uid, devices, reason, options = {}) {
       devices = filterSupportedDevices(options.data, devices);
-      const events = pushReasonsToEvents[reason];
-      if (!events) {
+      if (!PUSH_REASONS.has(reason)) {
         throw `Unknown push reason: ${reason}`;
       }
       // There's no spec-compliant way to error out as a result of having
       // too many devices to notify.  For now, just log metrics about it.
       if (devices.length > MAX_ACTIVE_DEVICES) {
-        reportPushError(new Error(ERR_TOO_MANY_DEVICES), uid, null);
+        log.warn(LOG_OP_TOO_MANY_DEVICES, { uid });
       }
       for (const device of devices) {
         const deviceId = device.id;
 
-        log.trace(LOG_OP_PUSH_TO_DEVICES, {
-          uid: uid,
-          deviceId: deviceId,
-          pushCallback: device.pushCallback,
-        });
+        // Details that we might want to include in logs.
+        const metricsTags = {
+          reason,
+          uid,
+          deviceId,
+          uaOS: device.uaOS,
+          uaOSVersion: device.uaOSVersion,
+          uaBrowser: device.uaBrowser,
+          uaBrowserVersion: device.uaBrowserVersion,
+        };
+        this.reportPushAttempt(device.pushCallback, metricsTags);
```

Comment:
This now reports an `attempt` for every device, even ones that we know are definitely going to fail due to e.g. a missing or expired callback. I think that will make analysing these events easier overall, as they form a more deliberate funnel.